chia/util/keychain.py

```diff
@@ -238,14 +242,15 @@ class Keychain:
     def __init__(self, user: Optional[str] = None, service: Optional[str] = None, force_legacy: bool = False):
         self.user = user if user is not None else default_keychain_user()
         self.service = service if service is not None else default_keychain_service()
-        if force_legacy:
-            legacy_keyring_wrapper = KeyringWrapper.get_legacy_instance()
-            if legacy_keyring_wrapper is not None:
-                self.keyring_wrapper = legacy_keyring_wrapper
-            else:
-                return None
-        else:
-            self.keyring_wrapper = KeyringWrapper.get_shared_instance()
+
+        keyring_wrapper: Optional[KeyringWrapper] = (
+            KeyringWrapper.get_legacy_instance() if force_legacy else KeyringWrapper.get_shared_instance()
+        )
+
+        if keyring_wrapper is None:
+            raise KeyringNotSet(f"KeyringWrapper not set: force_legacy={force_legacy}")
```

Comment:
This case isn't actually hit today, but I included it to help surface potential issues with future changes.

Reply:
The prior code assumed that returning `None` would help the caller detect an error condition. Since that doesn't work, raising an exception seems cleaner. The object isn't usable without a valid keyring_wrapper property.